File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num:nums){
            seen.add(num);
        }

        int res = 0;
        for (int num:nums){
            if (seen.contains(num-1)) continue;

            int cur = num, len = 0;
            while (seen.contains(cur)){
                len++;
                cur++;
            }
            res = Math.max(res, len);
        }
        return res;
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        res = len(nums)

        while l <= r:
            m = l + (r - l) // 2

            if nums[m] >= target:
                res = m
                r = m - 1
            else:
                l = m + 1
        
        return res
        